internal/controller/controllers/agent_controller.go

```diff
@@ -1250,6 +1250,30 @@ func (r *AgentReconciler) updateHostIgnition(ctx context.Context, log logrus.Fie
 	return err
 }
 
+func (r *AgentReconciler) updateNodeLabels(log logrus.FieldLogger, host *common.Host, agent *aiv1beta1.Agent, params *installer.V2UpdateHostParams) (bool, error) {
+	var err error
+	hostNodeLabels := make(map[string]string)
+	if host.NodeLabels != "" {
+		if err = json.Unmarshal([]byte(host.NodeLabels), &hostNodeLabels); err != nil {
+			log.WithError(err).Errorf("failed to unmarshal node labels for host %s infra-env %s", host.ID.String(), host.InfraEnvID.String())
+			return false, err
+		}
+	}
+	slicedNodeLabels := funk.Map(hostNodeLabels, func(key, value string) aiv1beta1.NodeLabel { return aiv1beta1.NodeLabel{Key: key, Value: value} })
+	if !funk.Subset(slicedNodeLabels, agent.Spec.NodeLabels) ||
```

Comment:
Why subset? I think it should fully match it no?

Reply:
Changed to map, so irrelevant